open-api/rest-catalog-open-api.yaml

```diff
@@ -3647,6 +4080,105 @@ components:
             type: integer
           description: "List of equality field IDs"
 
+    PlanTableScanRequest:
+      type: object
+      properties:
+        snapshot-id:
+          description:
+            Identifier for the snapshot to scan in a point-in-time scan
+          type: integer
+          format: int64
+        select:
+          description: List of selected schema fields
+          type: array
+          items:
+            $ref: '#/components/schemas/FieldName'
+        filter:
+          description:
+            Expression used to filter the table data
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Enables case sensitive field matching for filter and select
+          type: boolean
+          default: true
+        use-snapshot-schema:
+          description:
+            Whether to use the schema at the time the snapshot was written.
+
+            When time travelling, the snapshot schema should be used (true).
+            When scanning a branch, the table schema should be used (false).
+          type: boolean
+          default: false
+        start-snapshot-id:
+          description: Starting snapshot ID for an incremental scan (exclusive)
+          type: integer
+          format: int64
+        end-snapshot-id:
+          description:
+            Ending snapshot ID for an incremental scan (inclusive).
+
+            Required when start-snapshot-id is specified.
+          type: integer
+          format: int64
+        stats-fields:
+          description:
+            List of fields for which the service should send column stats.
+          type: array
+          items:
+            $ref: '#/components/schemas/FieldName'
+
+    FieldName:
+      description:
+        A full field name (including parent field names), such as those passed
+        in APIs like Java `Schema#findField(String name)`.
+
+        The nested field name follows these rules
+        - Nested struct fields are named by concatenating field names at each
+          struct level using dot (`.`) delimiter, e.g.
+          employer.contact_info.address.zip_code
+        - Nested fields in a map key are named using the keyword `key`, e.g.
+          employee_address_map.key.first_name
+        - Nested fields in a map value are named using the keyword `value`,
+          e.g. employee_address_map.value.zip_code
+        - Nested fields in a list are named using the keyword `element`, e.g.
+          employees.element.first_name
+      type: string
+
+    FetchScanTasksRequest:
+      type: object
+      required:
+        - plan-task
+      properties:
+        plan-task:
+          $ref: '#/components/schemas/PlanTask'
+
+    PlanTask:
+      description:
+        An opaque JSON object provided by the REST server that represents a
```

Comment:
Do we want this to be an Object or a String type?  Just looking at the generate python code, having an unstructured object definition could be difficult to work with and clear how other languages would deserialize it.

This should be opaque, but do we need to require it be an object?

Reply:
I would +1 for using a string. using an opaque object seems to be a bit difficult to parse and use, just after prototyping this at server side. There are also concerns from our security engineers for opening another injection surface directly in API model, that is not protected by either the service's data model layer, nor string parsers that are typically more easily reviewed. But these are all solvable issues. Curious what others think.

Reply:
@danielcweeks @jackye1995 @rdblue I am also ok with representing `PlanTask` as an `opaque string` as it is the most simple for both client and service to send/receive. I think it also offers more flexibility as for what the planTask can be. My only question then is we go with `string` are we being explicit in the description that this is a `opaque JSON string` or do we leave it up for service to provide anything for this?

Reply:
I'm +1 on string as well. Broadly that seems to me to be the easiest representation to work with across language implementations and will have just as much fidelity in what can be represented.

>My only question then is we go with string are we being explicit in the description that this is a opaque JSON string or do we leave it up for service to provide anything for this?

I think in the spec we should just be explicit in the description that this is opaque.

Reply:
+1 for string. I originally wanted to not specify a type at all, but I don't have a strong opinion and @danielcweeks says it will be easier for implementations.

Reply:
Will fix this to be string